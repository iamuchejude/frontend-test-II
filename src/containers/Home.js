import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import spotifyLogo from '../assets/images/spotify.png';
import giphyLogo from '../assets/images/giphy.png';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const url = `https://accounts.spotify.com/authorize?client_id=${process.env.SPOTIFY_CLIENT_ID}&scope=user-read-email%20user-top-read&response_type=token&redirect_uri=http://localhost:1234/callback`;
    return (
      <div id="home">
        <div className="content">
          <div className="intro">
            <p>Welcome, kindly click on any of the icons to navigate</p>
          </div>
          <div className="nav_items">
            <ul>
              <li>
                <a href={ url }>
                  <img src={ spotifyLogo } alt="Spotify" />
                </a>
              </li>
              <li>
                <Link to="/giphy">
                  <img src={ giphyLogo } alt="Giphy" />
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    )
  }
}

export default Home;
