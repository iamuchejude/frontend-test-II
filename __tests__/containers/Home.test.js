import React from 'react';
import { shallow } from 'enzyme';
import Home from '../../src/containers/Home';

describe('App components', () => {
  it('should render app', () => {
    const wrapper  = shallow(<Home />);

    expect(wrapper).toMatchSnapshot()
  })
});
