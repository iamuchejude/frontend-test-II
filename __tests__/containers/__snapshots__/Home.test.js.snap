// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components should render app 1`] = `
<div
  id="home"
>
  <div
    className="content"
  >
    <div
      className="intro"
    >
      <p>
        Welcome, kindly click on any of the icons to navigate
      </p>
    </div>
    <div
      className="nav_items"
    >
      <ul>
        <li>
          <a
            href="https://accounts.spotify.com/authorize?client_id=undefined&scope=user-read-email%20user-top-read&response_type=token&redirect_uri=http://localhost:1234/callback"
          >
            <img
              alt="Spotify"
              src={Object {}}
            />
          </a>
        </li>
        <li>
          <Link
            replace={false}
            to="/giphy"
          >
            <img
              alt="Giphy"
              src={Object {}}
            />
          </Link>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
